mediapipe/tasks/ios/audio/core/sources/MPPAudioRecord.m

```diff
@@ -165,7 +194,34 @@ - (nullable MPPFloatBuffer *)readAtOffset:(NSUInteger)offset
   return bufferToReturn;
 }
 
-- (void)startTappingMicrophoneWithError:(NSError **)error {
+- (BOOL)startTappingMicrophoneWithError:(NSError **)error {
+  // Stopping and resetting the audio engine to handle the case where user maybe resuming the audio
+  // engine stopped by the OS due to an interrupt (eg:, a phone call) or a route change. In such
+  // cases, if a new tap is installed without removing an existing tap the app will crash.
+  [self stopAndResetAudioEngine];
+
+  // For tapping the microphone, `AVAudioSession`'s `category` must be set and it must be
+  // activated. This audio record is not allowed to tap the microphone with any custom option
+  // including
+  // `.allowBluetooth`. This disallows  microphone route changes in most scenarios to ensure that
+  // the recording happens through the device microphone. If users need more control over the
+  // recording devices or OS interruptions, native `AVAudioEngine` can be used. `category` is set to
+  // `AVAudioSessionCategoryPlayAndRecord` to ensure that audio playback can be configured in the
+  // app while an audio record is running.
+  // TODO: Investigate safe starting of `AVAudioEngine` without any side effects to enable this
+  // class ot accept custom category, mode and options.
+  if (!([[AVAudioSession sharedInstance]
+            setCategory:AVAudioSessionCategoryPlayAndRecord
+                   mode:AVAudioSessionModeDefault
+                options:AVAudioSessionCategoryOptionOverrideMutedMicrophoneInterruption
```

Comment:
This is only available in 14.5. I added version guards but the internal tests are failing still (even though I bumped that version too). Is there something we can with iOS 12?